### ISSUE TITLE
[Feature request] Syntax highlighting in <script type="text/babel"> for HTML support

### ISSUE THREAD

**Issue by Kristinita** [babel/babel-sublime#291]
### Summary

It would be nice, if in `JavaScript (Babel)` syntax will be added support syntax highlighting for Babel in tag `<script></script>` for HTML.

### Expected behavior

In separate file.

![In separate file](http://i.imgur.com/d49VkkH.png)

### Actual behavior

In HTML.

![In HTML](http://i.imgur.com/CbU4qx0.png)

### Steps to reproduce

In open HTML file I have script:

```html
<script type="text/babel">
  let arr = ["hello", 2]; // let

  let [str, times] = arr; //

  alert( str.repeat(times) );
</script>
```

Script not highlight for me. I get scope for each place of this script: `source.js meta.jsx.js` I get each time.

### Environment

**Sublime Text:**
Build 3126
**Syntax:**
JavaScript (Babel) into Babel plugin
**Color scheme:**
Monokai Phoenix into Babel plugin

Thanks.

**Comment by nwoltman** [comment]
It looks like you're trying to set the syntax of HTML files to `JavaScript (Babel)`. Instead, you should use the `HTML` syntax for HTML files (while you have a `.html` file open, click `View` -> `Syntax` -> `Open all with current extension as...` -> `HTML`). Then in your Sublime settings, disable the native JavaScript package by adding `"JavaScript"` to the `"ignored_packages"` option.

i.e. In your `Preferences.sublime-settings` file
```json
{
  "ignored_packages":
  [
    "JavaScript",
    "Vintage"
  ]
}
```
By doing this, Babel will become the default JavaScript syntax highlighter, so it will highlight `<script>` tags in your HTML files.

**Comment by nwoltman** [comment]
You might have better luck opening an issue in the [sublimehq/Packages](https://github.com/sublimehq/Packages) repo since it should be the Sublime HTML syntax that includes this feature.

**Comment by zertosh** [comment]
Can't do anything about this. What the HTML grammar decides to scope as embedded javascript is outside babel-sublime.

**Comment by alopatin** [comment]
I was having this issue too, this is how I fixed it (on MacOS):

1. Copy `Sublime Text.app/Contents/MacOS/Packages/HTML.sublime-package` to somewhere.
2. Rename the file to `HTML.sublime-package.zip`, then unzip it.
3. Copy the contained `HTML.sublime-syntax` file to `~/Library/Application Support/Sublime Text 3/Packages/User/`.
4. Grep for `javascript`, change it to `javascript|babel`.
5. Close and reopen your original HTML file to see the updated syntax highlighting!

For convenience, [here is a gist](https://gist.github.com/alopatin/8f0d8f8c8e2d0c24bfbca8d0cbfa2a6a) with the modified `HTML.sublime-syntax` file that you can just copy to `~/Library/Application Support/Sublime Text 3/Packages/User/`.

**Comment by stanbers** [comment]
@alopatin , when I copy the HTML.sublime-syntax file to that path you mentioned above,  my sublime still not work ,did I miss anything ? do you have any suggestions ?

**Comment by asdutoit** [comment]
@stanbers , I assume you came right.  But just for the sake of anybody else who may be experiencing the same issue, rather save the file, `HTML.sublime-syntax`, in `~/Library/Application Support/Sublime Text 3/Packages/HTML/` .   Ensure Sublime Text 3 is closed before you start.
Then re-open Sublime and test.

**Comment by animex** [comment]
@alopatin Thanks. Did the same with Windows! The only difference is the line is different where you need to change in HTMl.sublime-syntax:
`  javascript_mime_type: |-
    (?ix)(?:
      # https://mimesniff.spec.whatwg.org/#javascript-mime-type
      (?:application|text)/(?:x-)?(?:java|ecma)script
      | text/javascript1\.[0-5]
      | text/babel
      | text/jscript
      | text/livescript
    )
`

**Comment by chavab1** [comment]
I finally got it to work half-way using @alopatin 's help. However, it still doesn't work entirely. Some of the declarations are all yellow after entering some JSX code. Do I have to set a setting somewhere??
Any idea how I can get it to work??
![capture](https://user-images.githubusercontent.com/6477242/47886013-f6f7e500-ddf4-11e8-9f0b-7c730adb17b0.PNG)


Thanx

**Comment by dtg5** [comment]
I have the same problem as @chavab1.
For me it looks like this:
![image](https://user-images.githubusercontent.com/22200244/47971760-66bed780-e062-11e8-8ef2-dbaefdcbbcb0.png)

**Comment by chavab1** [comment]
@dtg5 I actually figured out that it was the ending `/` that was causing the code to look that way. After I figured that out I did research and found that other people had started complaining about it as well and there is even a thread dedicated to it. It seems like support for JSX is still off.
They suggest to use [this](https://github.com/borela/naomi) syntax. It does work after installing this package. 

However, after enabling this new syntax I found that the PHP syntax doesn't work anymore. I researched that and the developers just became aware of it. They suggest to disable the default PHP syntax and use their own PHP syntax. They said it's just a temporary fix and they are currently working on a permanent solution.

After going through all that and then seeing your solution I'm kind of tempted to just use your solution. It's definitely much easier. Here's how my HTML looks now.

![capture](https://user-images.githubusercontent.com/6477242/48016259-01a9c700-e0e0-11e8-80df-caab9afaad76.PNG)

**Comment by woojoo666** [comment]
still isn't working for me, even with the solution from @dtg5. I know the syntax file is being loaded because it's parsing the 'text/babel' tag, but it's parsing it as plain javascript. Can anybody provide the sublime-syntax file that is working for them (Windows, Sublime Text 3)? Thanks

![img](https://i.imgur.com/UJQkp4D.png)

**Comment by chavab1** [comment]
It looks like you're still having the problem with the '/' sign.
As I stated in my previous response, I got it to work by using [this](https://github.com/borela/naomi) syntax. However, it conflicted with my PHP syntax so I had to use a different PHP syntax as well. 
Another solution was using making revisions to the main HTML syntax as @dtg5 discovered.
I've attached the file that I'm using and it's working. I made the changes that @dtg5 came up with. I'm using the [babel](https://packagecontrol.io/packages/Babel) package as well.

**Comment by chavab1** [comment]
It looks like you're still having the problem with the '/' sign.
As I stated in my previous response, I got it to work by using [this](https://github.com/borela/naomi) syntax. However, it conflicted with my PHP syntax so I had to use a different PHP syntax as well.

Another solution was making revisions to the main HTML syntax as @dtg5 discovered.

I've attached the file that I'm using with @dtg5's fix and it's working. I'm using the babel package as well.

[HTML.zip](https://github.com/babel/babel-sublime/files/3199911/HTML.zip)
